dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,371 @@
+                        ArrayList<UUID> newvals = new ArrayList<>();
+                        for(Object oval: vals) {
+                                try {
+                                        int legacy = Integer.parseInt(oval.toString());
+                                        UUID uuid = null;
+                                        if (col == FIELD.id) {
+                                                Object otype = input.getFieldValue("type");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else if (col == FIELD.scopeId) {
+                                                Object otype = input.getFieldValue("scopeType");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else {
+                                                uuid = mapId(col, legacy);
+                                        }
+                                        if (uuid != null) {
+                                                newvals.add(uuid);
+                                        }
+                                } catch (NumberFormatException e) {
+                                        log.warn("Non numeric legacy id "+ col.name() +":" + oval.toString());
+                                }
+                        }
+                        if (newvals.size() > 0) {
+                                input.removeField(col.name());
+                                for(UUID uuid: newvals) {
+                                        input.addField(col.name(), uuid.toString());
+                                }
+                                
+                        }
+                }
+        }
+        
+        private UUID mapId(FIELD col, int val) throws SQLException {
+                
+                if (col == FIELD.owningComm) {
+                        Community comm = communityService.findByLegacyId(context, val);
+                        return comm == null ? null : comm.getID();
+                }
+                if (col == FIELD.owningColl) {
+                        org.dspace.content.Collection coll = collectionService.findByLegacyId(context, val);
+                        return coll == null ? null : coll.getID();
+                }
+                if (col == FIELD.owningItem) {
+                        Item item = itemService.findByLegacyId(context, val);
+                        return item == null ? null : item.getID();
```

Comment:
I would keep Communities and Collections in the Hibernate cache (because there number is limited), but I would uncache all items. So that would be:

```
Item item = itemService.findByLegacyId(context, val);
UUID itemId = item == null ? null : item.getID();
context.uncacheEntity(item);
return itemId;
```